docs/coin_lifecycle.md

```diff
@@ -0,0 +1,119 @@
+---
+id: coin_lifecycle
+title: 7 - Lifecycle of a Coin
+---
+
+You should now understand how to create Chialisp puzzles that lock up coins with a particular set of rules.  You may be wondering how a coin gets locked up, where it is stored, how and when it gets spent, and who can spend it.  Let's take a look at the steps a coin goes through from its creation to its destruction.
+
+## The Coin Set Model
+
+Chia's model of how coins are stored is called the **coin set** model and is closely modelled after Bitcoin's UTXO model.  The idea is simple, every full node holds onto a database of **coin records**:
+
+```
+class Coin:
+  parent_coin_info: bytes32
+  puzzle_hash: bytes32
+  amount: uint64
+
+class CoinRecord:
+  coin: Coin
+  confirmed_block_index: uint32
+  spent_block_index: uint32
+  spent: bool
+  coinbase: bool
+  timestamp: uint64
+```
+
+This database is generated by looking at the blockchain from block height zero and processing all of the transactions until it is in sync with the current peak.  When a full node processes a spend, it first makes sure that the coin being spent exists in its database, runs and validates the Chialisp conditions that are produced, marks the coin that was being spent as spent, and adds any new coins that were created as a result.
+
+This approach makes the database very light as the blockchain grows, but still allows for complexity and smart contracts through Chialisp which is committed to inside the puzzle hash.  It is also an important distinction from some other blockchains in that *all state in Chialisp is stored in the coins*.  There is no special way to store data so that people can access it; the majority of that functionality comes from finding ways to reveal the proper information in puzzle reveals and by committing to predictable puzzle hashes.
```

Comment:
We should note that the puzzle and solutions are not stored

Reply:
Puzzles and solutions are stored in the transaction generator so I neglected to include this.